src/darwin/Framework/CHIP/MTRDeviceController_Concrete.mm

```diff
@@ -1409,12 +1409,12 @@ - (void)getSessionForNode:(chip::NodeId)nodeID completion:(MTRInternalDeviceConn
 
 - (void)directlyGetSessionForNode:(chip::NodeId)nodeID completion:(MTRInternalDeviceConnectionCallback)completion
 {
-    // TODO: Figure out whether the synchronization here makes sense.  What
-    // happens if this call happens mid-suspend or mid-resume?
     if (self.suspended) {
         MTR_LOG_ERROR("%@ suspended: can't get session for node %016llX-%016llx (%llu)", self, self.compressedFabricID.unsignedLongLongValue, nodeID, nodeID);
-        // TODO: Can we do a better error here?
-        completion(nullptr, chip::NullOptional, [MTRError errorForCHIPErrorCode:CHIP_ERROR_INCORRECT_STATE], nil);
+        dispatch_async(_chipWorkQueue, ^{
```

Comment:
So the thing is... asyncGetCommissionerOnMatterQueue can also sync-call completion, right?

Reply:
We need to either fix whoever is calling `directlyGetSessionForNode` to handle the fact that they might get a sync-call in whatever state they are in to the error handler, or change the API documentation for `getSessionForNode` and the implementations of both that and `directlyGetSessionForNode` to guarantee async callbacks.

We used to not be able to use the Matter queue there, because without a running controller we potentially had no running queue.  But now that we never shut the queue down (right?) maybe we can consistently deliver the errors on that queue....

Reply:
Can this be handled in a separate PR? @vivien-apple feel free to break this apart.

Reply:
Context: I'm working on aligning all our queues, and it's quite clear that our queue management is not aligned, and working as expected, I see queues being called back on the main queue for a moment, then on the delegate or work queue. Let's not boil the ocean in this fix, that one I'm handling separately.